smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -188,14 +211,51 @@ contract MixinPurchase is
     }
   }
 
+  /**
+  * Renew a given token
+  * @notice only works for non-free, expiring, ERC20 locks
+  * @param _tokenId the ID fo the token to renew
+  * @param _referrer the address of the person to be granted UDT
+  */
+  function renewMembershipFor(
+    uint _tokenId,
+    address _referrer
+  ) public {
+    _lockIsUpToDate();
+    _isKey(_tokenId);
+
+    // check the lock
+    require(_originalDurations[_tokenId] != type(uint).max, 'NON_EXPIRING_LOCK');
```

Comment:
maybe we can skip this to save on size?

Reply:
yes this will revert by overflow anyway as we will add to max uint. Let's keep it tho - explicit is always better